homeassistant/components/light/yeelight.py

```diff
@@ -6,157 +6,311 @@
 """
 import logging
 import socket
+import colorsys
 
 import voluptuous as vol
 
-from homeassistant.const import CONF_DEVICES, CONF_NAME
-from homeassistant.components.light import (ATTR_BRIGHTNESS, ATTR_RGB_COLOR,
-                                            ATTR_COLOR_TEMP,
-                                            SUPPORT_BRIGHTNESS,
-                                            SUPPORT_RGB_COLOR,
-                                            SUPPORT_COLOR_TEMP,
-                                            Light, PLATFORM_SCHEMA)
+from homeassistant.util.color import (
+    color_temperature_mired_to_kelvin as mired_to_kelvin,
+    color_temperature_kelvin_to_mired as kelvin_to_mired,
+    color_temperature_to_rgb)
+from homeassistant.const import CONF_DEVICES, CONF_NAME, STATE_OFF
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, ATTR_RGB_COLOR, ATTR_TRANSITION, ATTR_COLOR_TEMP,
+    ATTR_FLASH, FLASH_SHORT, FLASH_LONG,
+    SUPPORT_BRIGHTNESS, SUPPORT_RGB_COLOR, SUPPORT_TRANSITION,
+    SUPPORT_COLOR_TEMP, SUPPORT_FLASH,
+    Light, PLATFORM_SCHEMA)
 import homeassistant.helpers.config_validation as cv
-from homeassistant.util import color as color_util
-from homeassistant.util.color import \
-    color_temperature_mired_to_kelvin as mired_to_kelvin
 
-REQUIREMENTS = ['pyyeelight==1.0-beta']
+REQUIREMENTS = ['yeelight==0.1.0']
 
 _LOGGER = logging.getLogger(__name__)
 
-DOMAIN = 'yeelight'
+CONF_TRANSITION = "transition"
+DEFAULT_TRANSITION = 350
+
+CONF_SAVE_ON_CHANGE = "save_on_change"
+CONF_MODE_MUSIC = "use_music_mode"
 
-SUPPORT_YEELIGHT = (SUPPORT_BRIGHTNESS | SUPPORT_RGB_COLOR |
-                    SUPPORT_COLOR_TEMP)
+DOMAIN = 'yeelight'
 
-DEVICE_SCHEMA = vol.Schema({vol.Optional(CONF_NAME): cv.string, })
+DEVICE_SCHEMA = vol.Schema({
+    vol.Optional(CONF_NAME): cv.string,
+    vol.Required(CONF_TRANSITION, default=DEFAULT_TRANSITION): cv.positive_int,
```

Comment:
Oh, then it was a user error from my side (passing an empty dict directly instead of passing it through DEVICE_SCHEMA), as it wasn't working before. I'll check it out tomorrow and update accordingly.